stable_baselines3/common/buffers.py

```diff
@@ -396,3 +449,288 @@ def _get_samples(self, batch_inds: np.ndarray, env: Optional[VecNormalize] = Non
             self.returns[batch_inds].flatten(),
         )
         return RolloutBufferSamples(*tuple(map(self.to_torch, data)))
+
+
+class DictReplayBuffer(ReplayBuffer):
+    """
+    Replay buffer used in off-policy algorithms like SAC/TD3.
+
+    :param buffer_size: Max number of element in the buffer
+    :param observation_space: Observation space
+    :param action_space: Action space
+    :param device:
+    :param n_envs: Number of parallel environments
+    :param optimize_memory_usage: Enable a memory efficient variant
```

Comment:
I would drop `optimize_memory_usage` support for DictReplayBuffer, no?
It feels like it adds quite some complexity.

Reply:
I agree we should keep things as simple as possible first, and then later add any further optimizations/features once base implementations.

Reply:
I can default `optimize_memory_storage` to False for now for it to be implemented/verified later. Sound like a plan?